cognition_objects/pipeline_log.py

```diff
@@ -220,3 +221,63 @@ def update_to_new_diff_structure(
 
     if with_commit:
         general.commit()
+
+
+def get_error_and_time_elapsed_by_conversation_id(
+    project_id: str,
+    conversation_id: str,
+) -> List[CognitionPipelineLogs]:
+    grouped_message_logs = (
+        session.query(
+            CognitionMessage.id,
+            func.sum(CognitionPipelineLogs.time_elapsed).label("time_logs_elapsed"),
+            func.bool_or(CognitionPipelineLogs.has_error).label("logs_have_error"),
+        )
+        .join(CognitionMessage)
+        .join(CognitionConversation)
+        .filter(
+            CognitionPipelineLogs.project_id == project_id,
+            CognitionPipelineLogs.message_id == CognitionMessage.id,
+            CognitionMessage.conversation_id == conversation_id,
+        )
+        .group_by(CognitionMessage.id)
+        .all()
+    )
+    grouped_logs_dict = {}
+    for message in grouped_message_logs:
+        grouped_logs_dict[message.id] = {
+            "time_logs_elapsed": message.time_logs_elapsed,
+            "logs_have_error": message.logs_have_error,
+        }
+
+    return grouped_logs_dict
+
+
+def get_error_and_time_elapsed_by_conversation_ids(
+    project_id: str,
+    conversation_ids: List[str],
+) -> Dict[str, CognitionPipelineLogs]:
+    grouped_message_logs = (
+        session.query(
+            CognitionMessage.id,
+            func.sum(CognitionPipelineLogs.time_elapsed).label("time_logs_elapsed"),
+            func.bool_or(CognitionPipelineLogs.has_error).label("logs_have_error"),
+        )
+        .join(CognitionMessage)
```

Comment:
Not sure what is optimized by sqlalchemy but maybe print the full query to see how it would look (`general.print_orm_query`) because i suspect that the join with CognitionMessage on itself creates a query like
```
SELECT ...
FORM CognitionMessage
INNER JOIN CognitionMessage
INNER JOIN CognitionConversation
WHERE ...
```

which would be an unnecessary join on itself. The on criteria can btw be created like this:
![image](https://github.com/user-attachments/assets/c0f4d970-d068-4d3e-a99c-cd5e5ae48c17)